components/client/src/main/java/com/hotels/styx/client/StyxHttpClient.java

```diff
@@ -96,73 +156,165 @@ private static Origin originFromRequest(FullHttpRequest request, Boolean isHttps
         return newOriginBuilder(host.getHostText(), host.getPort()).build();
     }
 
+
     /**
      * Builder for {@link StyxHttpClient}.
      */
     public static class Builder {
-        private Connection.Factory connectionFactory;
-        private TlsSettings tlsSettings;
-        private String userAgent;
-        private ConnectionSettings connectionSettings = new ConnectionSettings(1000);
-        private int responseTimeout = 60000;
+        private String threadName = "simple-netty-http-client";
+        private int connectTimeoutMillis = 1000;
         private int maxResponseSize = 1024 * 100;
+        private int responseTimeout = 60000;
         private int maxHeaderSize = 8192;
-        private String threadName = "simple-netty-http-client";
+        private TlsSettings tlsSettings;
+        private boolean isHttps;
+        private String userAgent;
 
-        public Builder connectionSettings(ConnectionSettings connectionSettings) {
-            this.connectionSettings = connectionSettings;
-            return this;
+        public Builder() {
+        }
+
+        Builder(Builder another) {
+            this.threadName = another.threadName;
+            this.connectTimeoutMillis = another.connectTimeoutMillis;
+            this.maxResponseSize = another.maxResponseSize;
+            this.responseTimeout = another.responseTimeout;
+            this.maxHeaderSize = another.maxHeaderSize;
+            this.tlsSettings = another.tlsSettings;
+            this.isHttps = another.isHttps;
+            this.userAgent = another.userAgent;
         }
 
         /**
-         * Sets the user-agent header value to be included in requests.
+         * Sets a thread name used for the thread pool.
          *
-         * @param userAgent user-agent
-         * @return this builder
+         * @param threadName thread name
+         * @return this {@link Builder}
          */
-        public Builder userAgent(String userAgent) {
-            this.userAgent = userAgent;
+        public Builder threadName(String threadName) {
+            this.threadName = threadName;
             return this;
         }
 
-        public Builder responseTimeoutMillis(int responseTimeout) {
-            this.responseTimeout = responseTimeout;
+        /**
+         * Sets the TCP connection timeout.
+         *
+         * @param timeoutMs desired TCP connection timeout
+         * @return this {@link Builder}
+         */
+        public Builder connectTimeout(int timeoutMs) {
```

Comment:
We usually have a signature like `(int timeout, TimeUnit unit)` for these kinds of method.

Reply:
Yeah, this is a good suggestion.